src/components/LocalMap.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react'
+
+import { Observable } from 'rx'
+import { withStore } from 'fluorine-lib'
+
+import { decoding, constants } from 'pipboylib'
+const { parseBinaryMap, generateTreeFromDatabase } = decoding
+const { RequestLocalMapSnapshot } = constants.commands
+
+import {
+  SERVER_LOCALMAP_UPDATE
+} from '../constants/server_types'
+
+import PlayerArrow from './PlayerArrow'
+import dispatcher from '../dispatcher'
+import Database from '../stores/Database'
+
+const styles = {
+  localMap: {
+    display: 'block',
+    height: '100%',
+    width: '100%',
+    overflow: 'hidden'
+  },
+  canvas: {
+    background: '#000'
+  }
+}
+
+@withStore(dispatcher
+  .reduce(Database)
+  .map(x => generateTreeFromDatabase(x))
+  .map(x => x && x.Map.World.Player.Rotation)
+  .distinctUntilChanged(),
+  'orientation')
+@withStore(dispatcher
+  .filter(x => x && x.type === SERVER_LOCALMAP_UPDATE)
+  .map(x => parseBinaryMap(x.payload)),
+  'map')
+export default class LocalMap extends React.Component {
+  static contextTypes = {
+    sendCommand: React.PropTypes.func.isRequired
+  }
+
+  constructor(props, context) {
+    super(props, context)
+
+    this.state = {
+      width: props.map.width,
+      height: props.map.height
+    }
+  }
+
+  componentDidMount() {
+    this.updateMap(this.props.map)
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.map !== this.props.map) {
+      this.updateMap(nextProps.map)
+    }
+  }
+
+  updateMap = map => {
+    // Get next LocalMap
+    this.context.sendCommand('RequestLocalMapSnapshot')
```

Comment:
Since `updateMap` gets called in `componentDidMount`, this only requests the localmap when in view. Neat.